install/migrations/update_9.5.x_to_10.0.0/cable.php

```diff
@@ -0,0 +1,221 @@
+   $migration->addField('glpi_sockets', 'items_id', 'int', ['after' => 'itemtype']);
+   $migration->addKey("glpi_sockets", ['itemtype','items_id'], 'item');
+
+   $migration->addField('glpi_sockets', 'networkports_id', 'int', ['after' => 'items_id']);
+   $migration->addKey('glpi_sockets', 'networkports_id');
+
+   //migrate link between NetworkPort and Socket
+   // BEFORE : supported by NetworkPortEthernet / NetworkPortFiberchannel with 'sockets_id' foreign key
+   // AFTER  : supported by Socket with (itemtype, items_id, networkports_id)
+   $classes = [NetworkPortEthernet::getType(), NetworkPortFiberchannel::getType()];
+   foreach ($classes as $itemtype) {
+
+      $item = new $itemtype();
+      $datas = $item->find(['networkports_id'   => ['<>', 0],
+                            'netpoints_id'      => ['<>', 0]]);
+
+      foreach ($datas as $id => $values) {
+
+         //Load NetworkPort to get associated item
+         $networkport = new NetworkPort();
+         if ($networkport->getFromDB($values['networkports_id'])) {
+            $sockets_id = $values['netpoints_id'];
+            $socket = new Socket();
+            if ($socket->getFromDB($sockets_id)) {
+               $socket->update([
+                  'id'              => $sockets_id,
+                  'position'        => $networkport->fields['logical_number'],
+                  'itemtype'        => $networkport->fields['itemtype'],
+                  'items_id'        => $networkport->fields['items_id'],
+                  'networkports_id' => $networkport->getID()
+               ]);
+            }
+         }
+      }
+   }
+
+   //remove "useless "netpoints_id" field
+   $migration->dropField('glpi_networkportethernets', 'netpoints_id');
+   $migration->dropField('glpi_networkportfiberchannels', 'netpoints_id');
+   $migration->dropKey('glpi_networkportethernets', 'netpoint');
+   $migration->dropKey('glpi_networkportfiberchannels', 'netpoint');
+}
+
+if (!$DB->tableExists('glpi_networkportfiberchanneltypes')) {
+   $query = "CREATE TABLE `glpi_networkportfiberchanneltypes` (
+      `id` int NOT NULL AUTO_INCREMENT,
+      `name` varchar(255) DEFAULT NULL,
+      `comment` text,
+      `date_mod` timestamp NULL DEFAULT NULL,
+      `date_creation` timestamp NULL DEFAULT NULL,
+      PRIMARY KEY (`id`),
+      KEY `name` (`name`),
+      KEY `date_mod` (`date_mod`),
+      KEY `date_creation` (`date_creation`)
+      ) ENGINE = InnoDB DEFAULT CHARSET = {$default_charset} COLLATE = {$default_collation} ROW_FORMAT=DYNAMIC;";
+   $DB->queryOrDie($query, "10.0 add table glpi_networkportfiberchanneltypes");
+}
+
+$migration->addField('glpi_networkportfiberchannels', 'networkportfiberchanneltypes_id', 'int', ['after' => 'items_devicenetworkcards_id']);
+$migration->addKey('glpi_networkportfiberchannels', 'networkportfiberchanneltypes_id', 'type');
+
+$ADDTODISPLAYPREF['Socket'] = [5, 6, 9, 8, 7];
+$ADDTODISPLAYPREF['Cable'] = [4, 31, 6, 15, 24, 8, 10, 13, 14];
```

Comment:
This list is different thant the one in empty-data.sql.

Reply:
done